internal/app/document_store.go

```diff
@@ -0,0 +1,109 @@
+package app
+
+import (
+	"context"
+	"errors"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/dynamo"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/page"
+)
+
+type documentStore struct {
+	dynamoClient DynamoClient
+	s3Client     S3Client
+}
+
+func NewDocumentStore(dynamoClient DynamoClient, s3Client S3Client) DocumentStore {
+	return documentStore{dynamoClient: dynamoClient, s3Client: s3Client}
+}
+
+func (s documentStore) GetAll(ctx context.Context) (page.Documents, error) {
+	data, err := page.SessionDataFromContext(ctx)
+	if err != nil {
+		return []page.Document{}, err
```

Comment:
nothing major, but you should be able to return `nil` instead of `[]page.Document{}` for these